rkt/api_service.go

```diff
@@ -114,16 +114,9 @@ func isPartOf(keyword, s string) bool {
 	return strings.Contains(s, keyword)
 }
 
-// satisfiesPodFilter returns true if the pod satisfies the filter or the filter
-// is nil, which means the pod will be returned.
-// It returns false if the pod doesn't satisfy the filter, which means it will
-// not be returned.
+// satisfiesPodFilter returns true if the pod satisfies the filter.
+// The pod, manifest, filter must not be nil.
```

Comment:
> Well, I prefer to pass pointers when the struct is big.

Do not worry about that if the func is not in a tight loop or an a hot path like this one. And the struct is not HUGE either. I would worry about it when you have 10s of fields in a struct probably.

And, yes, if you want this func to be safe without additional assumption then you have to check if the pointer is nil.